lib/ansible/modules/remote_management/spacewalk/spacewalk_swchannel_facts.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, Davide Blasi (@davegarath)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: spacewalk_swchannel_facts
+short_description: This module can be used to gather facts about all channels in spacewalk
+description:
+  - "Gather facts about all channels in spacewalk"
+version_added: "2.7"
+author:
+  - Davide Blasi (@davegarath)
+extends_documentation_fragment: spacewalk.documentation
+'''
+
+EXAMPLES = '''
+# Pass in a message
+- name: Get all channels in spacewalk
+  spacewalk_swchannel_facts:
+      url: "{{ spacewalk_url_api }}"
+      login: "{{ spacewalk_login }}"
+      password: "{{ spacewalk_passowrd }}"
+  delegate_to: localhost
+  register: channels
+'''
```

Comment:
Perhaps add a `debug` task showing one the returned data?

Reply:
Added debug task